test/pylib/pool.py

```diff
@@ -38,6 +38,23 @@ async def start_server():
                 await pool.steal()
             else:
                 await pool.put(server)
+
+    To atomically free up space in the pool and use this space to obtain a new
+    object, you can use `steal_and_get`. This is different from a `steal` call
```

Comment:
Can we please drop "steal" at least if we have "steal and get"? I think we only had one use case for steal, so let's replace it with steal_and_get(). Also steal_ is a non-obvious reference in the context of "steal and get". Are we stealing the same thing as we are getting or different things? Perhaps get_and_steal() would sound better, but that would imply we're stealing what we're getting. Anyway, forget my bikeshedding on the name, I am just unhappy we have to use such a complex paradigm.

Reply:
Instead of guessing you could just read the docstring for `steal_and_get` which says:
```
Atomically take ownership of a previously borrowed object
           which frees up space in the pool, and borrow a new object.
```

Reply:
There is another use case for steal() in test.py: PythonTest.run